dds/DdsDcpsCore.idl

```diff
@@ -617,6 +617,8 @@ module OpenDDS {
     struct InternalThreadBuiltinTopicData {
       BUILT_IN_TOPIC_KEY string thread_id;
       DDS::OctetArray16 participant_guid;
+      double utilization;
+      DDS::Time_t timestamp;
```

Comment:
```suggestion
```

There actually already was a timestamp field in this struct before, but we decided to remove it because `SampleInfo::source_timestamp` is a way DDS already provides to get timestamp of a sample. https://github.com/objectcomputing/OpenDDS/pull/2247/commits/ac92e2dd1586b6babc2158bcfe9fea9b1a835631#diff-f32cd34225dfd7aaabc590e1494701202f10cd801600b0f37c9da8ff16113359

If you look at the `store_synthetic_data`, the timestamp is getting passed in, so it's the same value.

Reply:
Yes, I am aware of the timestamp you reference,  But isn't that the time the message was created/sent" My timestamp is the time the utilization value was measured.  it seems like these BIT updates happen at their own frequency which may be somewhat lower than the thread utilization freequency. So it may be the case that I need to extend this struct further to include a sequence fo double/time_t pairs. OTOH if  if a BIT sample is emitted as a result of any thread state update meaning there will never be more than one data point possible then I suppose this timestamp is redundant

Reply:
As part of the changes I linked to above I changed `store_synthetic_data` to optionally take the time stamp to use instead of doing the time at the moment like it normally does. For the thread BIT, it's the same value as what's in the `Thread::timestamp`, which for the thread status update is when the thread is declaring it's alive.

https://github.com/objectcomputing/OpenDDS/blob/fb0d13e99e0e148fde34473db22d7b8d545635c4/dds/DCPS/RTPS/Spdp.cpp#L4461-L4464

As for what the requirements of this are, yeah they might warrant something else than what was needed for thread status updates. The thread status was designed just to deliver the last update time. I don't know what it would look like or if it would actually be ideal, but you could aggregate multiple utilization into one. You should also be able to adjust the `thread_status_interval` since if need be.

Reply:
got it. Thanks for the explanation.